### PR TITLE
[TACHYON-590] Add unit tests for getLeadingPath in TachyonURI (@uronc…

### PR DESCRIPTION
**Pull request by tianyin**
…e-cc's comment)

@uronce-cc I added the unit tests for the added function in TachyonURI. Could you please review it for me? Thanks a lot!

~t

### CODE REVIEW

**Comment by uronce-cc** [comment]
+1 LGTM

File: common/src/test/java/tachyon/TachyonURITest.java [hunk 1]
```diff
@@ -483,4 +483,14 @@ private void testParentChild(String target, String parent, String child) {
           new TachyonURI(new TachyonURI(parent), new TachyonURI(child)).toString());
     }
   }
+  
+  @Test
+  public void testGetLeadingPath() {
```
Comment (apc999) [inline common/src/test/java/tachyon/TachyonURITest.java]: The naming convention in Tachyon should be `getLeadingPathTest` for this test case.

**Comment by apc999** [comment]
One minor comment left. Thanks!

File: common/src/test/java/tachyon/TachyonURITest.java (hunk 1 above)
Comment (tianyin) [inline common/src/test/java/tachyon/TachyonURITest.java]: Thanks a lot, Bin! I will follow this naming convention for all the unit test cases.

~t

File: common/src/test/java/tachyon/TachyonURITest.java [hunk 2]
```diff
@@ -483,4 +483,14 @@ private void testParentChild(String target, String parent, String child) {
...
   }
+  
+  @Test
+  public void getLeadingPathTest() {
+    Assert.assertEquals("/",      new TachyonURI("/a/b/c/").getLeadingPath(0));
+    Assert.assertEquals("/a",     new TachyonURI("/a/b/c/").getLeadingPath(1));
+    Assert.assertEquals("/a/b",   new TachyonURI("/a/b/c/").getLeadingPath(2));
+    Assert.assertEquals("/a/b/c", new TachyonURI("/a/b/c/").getLeadingPath(3));
+    Assert.assertEquals(null,     new TachyonURI("/a/b/c/").getLeadingPath(4));
+    Assert.assertEquals("/",      new TachyonURI("/").getLeadingPath(0));
```
Comment (gpang) [inline common/src/test/java/tachyon/TachyonURITest.java]: What about some other cases such as:
"" (empty string)
"."
"a/b/c"

**Comment by gpang** [comment]
I just had a minor question about additional test cases. Thanks for the PR!

**Comment by tianyin** [comment]
@gpang , wow, this is not a minor question, but a big one. 

My previous assumption is that TachyonURI always represents an absolute path pointing to a file in Tachyon namespace, e.g., 
"/a/b/c"
or 
"C:\a\b\b"
(as we will normalize the path when creating the URIs) 

You point me that TachyonURI can also be something like "", ".", "a/b/c". In these cases, what files the URI points to? @_@ It seems the getLeadingPath() function cannot deal with these paths, as the semantic is undefined...

**Comment by gpang** [comment]
O, I didn't mean for it to be a major question. I just looked at some other ways to create a TachyonURI like in: isRootTests() and joinTests(). If TachyonURI just normalizes any input, then I think everything should still work, right?

**Comment by tianyin** [comment]
so the paths such as "", ".", "a/b/c" can be generated into TachyonURI. These paths kind of breaks the semantics of functions such as getDepth() and getLeadingPath()

For example, the input `"."` represents the current directory. But if you call `getDepth()`/`getLeadingPath()`, you would get `0`/`null` because it is not translated into the absolute path.

**Comment by gpang** [comment]
I'm not sure what getLeadingPath() is used for, but does it make sense for it to return null in those cases?

**Comment by tianyin** [comment]
Yes, currently "" (empty string) and "." returns null, but "a/b/c" returns the "leading path." 

I will definitely add the test cases (thanks, @gpang !

I guess my question is more general as the abstraction of TachyonURI, and let me ask on the dev mailing list :-)

**Comment by tianyin** [comment]
@gpang I got it. Basically the TachyonURI of Tachyon file is always in the form of "/a/b/c" since we use a Unix-style namespace for Tachyon FS. But as a URI, it could identify any name so "" (empty string) and "C:\a" should also be valid. 

Test cases are added. Could you please review it?

**Comment by gpang** [comment]
Thanks for the info!

**Comment by gpang** [comment]
LGTM

**Pull request merged**